### PR TITLE
67 wp4 onnx inference benchmarking script solve onnxpy

### LINKED ISSUE #67: [WP4] ONNX Inference & Benchmarking Script (solve_onnx.py)
# [WP4] ONNX Inference & Benchmarking Script (`solve_onnx.py`)

## Description

Create a standalone ONNX inference script that runs trained SimplifiedHRM
models (exported via `export_onnx.py`) on CPU using only ONNX Runtime
**no PyTorch dependency required**. This is the key deployment script for
Raspberry Pi 5 and other resource-constrained devices.

The script must support all three puzzle types (sudoku_4x4, sudoku_9x9, maze)
and provide four modes: random demo, dataset evaluation, latency benchmarking,
and interactive solving.

### Why this matters
- RPi5 cannot run PyTorch efficiently (too heavy, no CUDA)
- ONNX Runtime has prebuilt aarch64 wheels for Bookworm
- This script is the **viva demo**: load model → solve puzzle → show result

## Tasks

- [ ] Implement `ONNXSolver` class (loads `.onnx`, runs inference via ORT)
- [ ] Add puzzle config for all 3 types (sudoku_4x4, sudoku_9x9, maze)
  - Maze needs `--maze-size` flag for grid dimensions
- [ ] Implement 4 modes:
  - **Demo mode** (default): generate random puzzles, solve, display with colour
  - **Dataset eval**: load `.npz`, compute token/puzzle accuracy, show examples
  - **Benchmark**: warmup + timed runs, report mean/std/min/max/throughput
  - **Interactive**: user types puzzle values, get instant solution
- [ ] Maze-specific display and metrics:
  - Format maze grid with wall/path/start/goal symbols
  - Path accuracy (on path cells only), cell accuracy (all cells)
  - Colour overlay: green=correct path, red=wrong, cyan=missed
- [ ] Sudoku display with box lines and colour coding
- [ ] Platform info in benchmark output (arch, processor, ORT version)
- [ ] Zero PyTorch dependency — only `numpy` and `onnxruntime`

## Acceptance Criteria

- [ ] `solve_onnx.py` runs on a machine with only `numpy` + `onnxruntime` installed (no torch)
- [ ] All 3 puzzle types work: `--puzzle sudoku_4x4 | sudoku_9x9 | maze`
- [ ] `--benchmark` prints latency table with mean, std, min, max, throughput
- [ ] `--data` evaluates on `.npz` and prints token accuracy + puzzle accuracy
- [ ] `--interactive` accepts comma-separated input and returns predictions
- [ ] Maze mode displays path overlay with colour coding
- [ ] File runs successfully on Raspberry Pi 5 (aarch64, Bookworm Lite)

## Commands

```bash
# Demo — solve random puzzles
python scripts/solve_onnx.py \
    --model model/simplified_hrm_sudoku_9x9.onnx --puzzle sudoku_9x9

python scripts/solve_onnx.py \
    --model model/simplified_hrm_maze_11x11_s4.onnx --puzzle maze --maze-size 11

# Evaluate on dataset
python scripts/solve_onnx.py \
    --model model/simplified_hrm_sudoku_9x9.onnx --puzzle sudoku_9x9 \
    --data data/sudoku_9x9_train.npz

python scripts/solve_onnx.py \
    --model model/simplified_hrm_maze_11x11_s4.onnx --puzzle maze \
    --maze-size 11 --data data/maze_11x11_train.npz

# Benchmark latency
python scripts/solve_onnx.py \
    --model model/simplified_hrm_sudoku_9x9.onnx --puzzle sudoku_9x9 --benchmark

python scripts/solve_onnx.py \
    --model model/simplified_hrm_maze_11x11_s4.onnx --puzzle maze \
    --maze-size 11 --benchmark

# Interactive mode
python scripts/solve_onnx.py \
    --model model/simplified_hrm_sudoku_9x9.onnx --puzzle sudoku_9x9 --interactive

# Compare step variants
python scripts/solve_onnx.py --model model/simplified_hrm_9x9_s16.onnx --puzzle sudoku_9x9 --benchmark
python scripts/solve_onnx.py --model model/simplified_hrm_9x9_s8.onnx  --puzzle sudoku_9x9 --benchmark
python scripts/solve_onnx.py --model model/simplified_hrm_9x9_s4.onnx  --puzzle sudoku_9x9 --benchmark
```

### PR DESCRIPTION
**Pull request by KyryloKozlovskyi**
[WP4] ONNX Inference & Benchmarking Script (solve_onnx.py) [#67 ](https://github.com/fionntmcc/cross-platform-hrm/issues/67)

### CODE REVIEW

**Review by fionntmcc — APPROVED** [review]
All good.
